hms-app/src/pages/YourIdeas.tsx

```diff
@@ -1,235 +1,93 @@
-
-  useEffect(() => {
-    setRelevantIdeaList((relevantIdeaList) => {
-      return []
-    })
-    loadRelevantIdeaDetails()
-  }, [hackathonData])
-
-  useEffect(() => {
-    if (
-      !relevantIdeaList
-        .map((relevant) => {
-          return relevant.id
-        })
-        .includes(ideaData.id)
-    ) {
-      setRelevantIdeaList((relevantIdeaList) => {
-        return [...relevantIdeaList, ideaData]
-      })
-    }
-  }, [ideaData])
 
   const filteredIdeas = relevantIdeaList.filter((item) => {
     return item.owner?.id.includes(userId)
   })
 
-  const data = hackathonList.hackathons.map(
-    (hackathon, index) => hackathon.title
-  )
+  const getHackathonID = (hackthonID: string) => {
+    setSelectedHackweek(hackthonID)
+  }
 
-  const selectChange = (value: string) => {
-    const getHackathon = hackathonList.hackathons.filter((hackathon) => {
-      return hackathon.title.includes(value)
-    })
+  const getRelevantIdeaList = (relevantIdeaList: Idea[]) => {
+    setRelevantIdeaList(relevantIdeaList)
+  }
 
-    const selectedHackathonID = getHackathon.map(
-      (hackathon, index) => hackathon.id
-    )
+  const getHackathonData = (hackathonData: Hackathon) => {
+    setHackathonData(hackathonData)
+  }
 
-    setSelectedHackweek(selectedHackathonID.toString())
+  const getIsLoading = (isLoading: boolean) => {
+    setIsLoading(isLoading)
   }
 
   return (
     <>
-      <h1>this is the your idea page</h1>
-      {hackathonList.isLoadingHackathonList && (
-        <div>hackathon select is loading...</div>
-      )}
-      {!hackathonList.isLoadingHackathonList &&
-        !hackathonList.errorHackathonList && (
-          <div>
-            <div style={{ width: 250 }}>
-              <Select
-                placeholder={'select a Hackathon'}
-                maxDropdownHeight={280}
-                data={data}
-                onChange={selectChange}
-              />
-            </div>
-            <Accordion mb={30} pt={10} iconPosition="left">
-              <Accordion.Item
-                style={{ border: '1px solid' }}
-                label={'Create new idea'}
-              >
-                <IdeaForm
-                  ideaID={'null'}
-                  hackathon={hackathonData}
-                  userId={userId}
-                  context={'new'}
-                />
-              </Accordion.Item>
-            </Accordion>
-          </div>
-        )}
+      <HackathonSelectDropdown setHackathonID={getHackathonID} />
 
-      {isHackathonError && (
-        <div>
-          <h3>Error loading hackathons</h3>
-          <p>something went wrong.</p>
-        </div>
-      )}
-      {isHackathonLoading && (
-        <div>
-          <h3>Hackathon details are loading...</h3>
-        </div>
-      )}
+      <RelevantIdeasLoader
+        setHackathon={getHackathonData}
+        setRelevantIdea={getRelevantIdeaList}
+        selectedHackweek={selectedHackweek}
+        setLoading={getIsLoading}
+      />
```

Comment:
so this should probably just be a function that is called outside of the html elements and not put here